components/compliance-service/inspec-agent/runner/runner.go

```diff
@@ -293,6 +293,27 @@ type ResolveTask struct {
 	resolver            *resolver.Resolver
 }
 
+func (t *ResolveTask) handleEmptyNodeJobs(job *jobs.Job) error {
+	now := time.Now()
+	nodeJob := &types.InspecJob{
+		InspecBaseJob: types.InspecBaseJob{
+			JobID: job.Id,
+		},
+		StartTime:  &now,
+		EndTime:    &now,
+		NodeStatus: types.StatusAborted,
+	}
+	err := t.scanner.UpdateJobStatus(nodeJob.JobID, "failed", nodeJob.StartTime, nodeJob.EndTime)
+	if err != nil {
+		logrus.Errorf("error updating job status for job %s (%s): %s", job.Name, job.Id, err.Error())
+	}
+	err = t.scanner.UpdateResult(context.TODO(), nodeJob, nil, &inspec.Error{Message: "no nodes found"}, "")
```

Comment:
The method this is called from is `Run(ctx context.Context, task cereal.Task)` so perhaps we can thread that context through rather than using context.TODO